packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js

```diff
@@ -58,18 +58,33 @@ angular.module('App').controller(
              ============================== */
       this.loading.availableRam = true;
 
-      this.privateDatabaseService.listAvailableRam().then((availableRam) => {
-        this.loading.availableRam = false;
-        this.data.availableRam = availableRam;
-        if (this.database.infrastructure === 'legacy') {
-          remove(this.data.availableRam, (ram) => ram === '2048');
-        }
-
-        remove(
-          this.data.availableRam,
-          (ram) => +ram === +this.database.ram.value,
-        );
-      });
+      this.privateDatabaseService
+        .getUpgradePlans(this.productId)
+        .then((result) => {
+          this.data.availableRam = result.map(
+            (e) => e.planCode.match(/[0-9]+/)?.[0],
```

Comment:
Could we get this information by something less tricky ? Like a product description, technical blob or whatever ?

Reply:
thanks for the review. Initially, I wanted to use constants to match plan code and ram, but dynamic extraction was suggested by Yann (see his comment in https://github.com/ovh/manager/pull/7899)